### PR TITLE
#10510 new Docker image for building against master branch

### PR DESCRIPTION
**Pull request by tjstavenger-pnnl**
This is a Docker image that has worked for PNNL so far in building against master from GitlabCI.

The ``locales`` dependency and ``RUN`` work seems like a kludge to get around perl warnings when running ``start_test``. I'm not sure how this was handled with the previous releases?

I added ``curl`` as an install so that the ``pip`` calls in ``make test-venv`` work. I was getting errors without it saying that ``curl`` didn't exist.

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@awallace-cray - want to review?

File: util/dockerfiles/master/gasnet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM chapel/chapel:master
```
Comment (awallace-cray) [inline util/dockerfiles/master/gasnet/Dockerfile]: Not a complaint, but whether this FROM line would work as intended depends on how the master-branch non-gasnet Chapel Docker image gets named after you build it; that image name is not defined in its Dockerfile. Instead, you name the image after you build it, with another `docker` command. At least, that's how I learned it.

Comment (tjstavenger-pnnl) [reply util/dockerfiles/master/gasnet/Dockerfile]: I copied https://github.com/chapel-lang/chapel/blob/master/util/dockerfiles/1.17.0/Dockerfile and it doesn't to a ``make cleanall`` either? A quick scan of 1.16 and 1.15 don't do it either.

The ``chapel/chapel:master`` was a copy & edit from 1.17 as well -- I just renamed ``latest`` to ``master``

FWIW -- I've only run the non-gasnet version to make https://hub.docker.com/r/pnnl/chapel/

Comment (tjstavenger-pnnl) [reply util/dockerfiles/master/gasnet/Dockerfile]: Ah, I see what I did. The cleanall isn't in the Dockerfiles in the versioned subfolders but is in the root.

**Comment by ben-albrecht** [comment]
@awallace-cray - Is anything blocking this PR from merging? Based on comments, I couldn't tell if we're waiting for a change from @tjstavenger-pnnl or not.

**Comment by tjstavenger-pnnl** [comment]
I believe it is ready to go.

**Comment by awallace-cray** [comment]
I agree. Thought I approved it already

**Comment by awallace-cray** [comment]
Oh, should I merge it then?

**Comment by awallace-cray** [comment]
ben-albrecht agrees this is okay to merge. As noted before, these files are purely advisory. 
Thanks!

**Comment by bradcray** [comment]
@awallace-cray / @ben-albrecht : It looks like we don't have a CLA for @tjstavenger-pnnl , is that correct?

**Comment by awallace-cray** [comment]
So it would seem.

**Comment by tjstavenger-pnnl** [comment]
Sorry for missing the CLA step -- how would you like me to proceed?

**Comment by bradcray** [comment]
Tim: No problem, it's our mistake for not catching it (we've intended to set up a bot to auto-flag PRs that need them, but...).

The right way to do this would be to have the appropriate person (director?  lawyer?  ...?) at PNNL fill out the Apache CCLA for our project, naming you as able to contribute (and, to prefetch, you could have them add names of other likely contributors).  I know this used to give the labs a conniption fit, but I think things have gotten better as Apache has become better known and more standardized.  More info here:  https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/

**Comment by tjstavenger-pnnl** [comment]
I'll ask our legal department about how to get the CCLA completed.

**Comment by bradcray** [comment]
Thanks Tim!  I think we could think of this as a de minimus change, but given that we'd like to get more code from you guys over time, I think it makes sense to see whether we can get a CCLA on file.

**Pull request merged**